src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -502,6 +533,21 @@ private async Task CreateExecutablesAsync(IEnumerable<AppResource> executableRes
                 }
 
                 await createResource().ConfigureAwait(false);
+
+                // NOTE: This check is only necessary for the inner loop in the dotnet/aspire repo. When
+                //       running in the dotnet/aspire repo we will normally launch the dashboard via
+                //       AddProject<T>. When doing this we make sure that the dashboard is running.
+                if (er.ModelResource.Name.Equals(KnownResourceNames.AspireDashboard, StringComparisons.ResourceName))
+                {
+                    if (er.ModelResource.TryGetAllocatedEndPoints(out var allocatedEndpoints))
+                    {
+                        // We just check the HTTP endpoint because this will prove that the dashboard
+                        // is listening and is ready to process requests.
+                        var httpEndpoint = allocatedEndpoints.Single(ae => ae.Name.Equals("http", StringComparisons.EndpointAnnotationName));
+                        await WaitForHttpSuccessOrThrow(httpEndpoint.UriString, TimeSpan.FromSeconds(DashboardWaitTimeInSeconds), cancellationToken).ConfigureAwait(false);
+                    }
+                }
```

Comment:
I wonder if we could condition all our "hacks" for developing in dotnet/aspire with `#if DEBUG`, so that they don't ship. Could use a different symbol if we wanted to launch release builds in VS too. Not something for this PR. Just an observation.

Reply:
Yeah I was thinking the same thing. I'm struggling to think of a reason why we couldn't.